src/printer.js

```diff
@@ -3658,14 +3666,16 @@ function isTypeAnnotationAFunction(node) {
   );
 }
 
-function isFlowNodeStartingWithDeclare(node, options) {
-  if (options.parser !== "flow") {
+function isNodeStartingWithDeclare(node, options) {
+  if (!(options.parser === "flow" || options.parser === "typescript")) {
```

Comment:
Would be nice if the ast could provide this information instead of having to go to the source for it. cc @mroch, @JamesHenry